src/Datadog.Trace/Tracer.cs

```diff
@@ -361,6 +366,130 @@ private static IStatsd CreateDogStatsdClient(TracerSettings settings, string ser
             return new Statsd(statsdUdp, new RandomGenerator(), new StopWatchFactory(), prefix: string.Empty, constantTags);
         }
 
+        private static void StartStandaloneAgentProcessesWhenConfigured()
+        {
+            try
+            {
+                var datadogConfigPath = Environment.GetEnvironmentVariable(ConfigurationKeys.AgentConfigPath);
+                string traceProcessArgs = null;
+                string dogStatsDArgs = null;
+
+                if (!string.IsNullOrWhiteSpace(datadogConfigPath))
+                {
+                    traceProcessArgs = $"--config \"{datadogConfigPath}\"";
+                }
+
+                _traceAgentMonitor = StartProcessWithKeepAlive(ConfigurationKeys.AgentPath, traceProcessArgs);
+                _dogStatsDMonitor = StartProcessWithKeepAlive(ConfigurationKeys.DogStatsDPath, dogStatsDArgs);
+            }
+            catch (Exception ex)
+            {
+                DatadogLogging.RegisterStartupLog(log => log.Error(ex, "Error when attempting to start standalone agent processes."));
+            }
+        }
+
+        private static bool ProgramIsRunning(string fullPath)
+        {
+            if (string.IsNullOrWhiteSpace(fullPath))
+            {
+                return false;
+            }
+
+            var fileName = Path.GetFileNameWithoutExtension(fullPath).ToLower();
+            var processesByName = Process.GetProcessesByName(fileName);
+
+            if (processesByName.Length > 0)
```

Comment:
Since you're getting filename without extension above, `processesByName` should not ever be null. I think, given that seemed to be this person's issue: https://stackoverflow.com/questions/15923371/why-is-the-process-getprocessesbyname-always-null 

But I wonder if it wouldn't be safer to just verify `processesByName` is not null before de-referencing?